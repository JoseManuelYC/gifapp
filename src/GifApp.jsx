import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifApp = () => {
  const [categories, setCategories] = useState(["Resident Evil"]);

  const onAddNewCategory = (newCategory) => {
    if (categories.includes(newCategory)) return; //para que no sea la misma categoria
    setCategories([newCategory, ...categories]);
    /*  setCategories((cat)=> [...categories, "Valorant"]); */
    /* setCategories(categories.concat("valorant")); */
  };

  return (
    <div className="container">
      <h1>GifApp</h1>
      <AddCategory onNewCategory={onAddNewCategory} />
      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </div>
  );
};
