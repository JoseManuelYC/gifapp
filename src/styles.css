* {
  font-family: Helvetica, Arial, sans-serif;
}

body {
  padding: 60px;
}

.container {
  background-color: rgb(21, 0, 138);
  padding: 20px;
  color: #fff;
  border-radius: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.subcontainer {
  display: flex;
  flex-direction: column;
  align-items: center;
}

input {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgb(97, 32, 158);
  color: black;
  font-size: 1.2rem;
  outline: none;
  padding: 10px 15px;
  width: 500px;
}

h2 {
  font-size: 1rem;
}

h3 {
  font-size: 2rem;
  margin-bottom: 5px;
}

.card-grid {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.card {
  align-content: center;
  align-items: center;
  border-radius: 10px;
  box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
  display: flex;
  flex-direction: column;
  height: 0%;
  justify-content: center;
  margin-bottom: 20px;
  margin-right: 20px;
  overflow: hidden;
}

.card p {
  border-radius: 10px;
  background-color: white;
  flex: 1;
  font-size: 1.5rem;
  margin-top: 5px;
  padding: 5px 20px 0px 20px;
  text-align: center;
  color: black;
}

.card img {
  width: 100%;
}
h1 {
  font-size: 3rem;
}
