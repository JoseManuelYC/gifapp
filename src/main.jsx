import React from "react";
import ReactDOM from "react-dom/client";
import { GifApp } from "./GifApp";

import "./styles.css";
// pruena 2
ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <GifApp />
  </React.StrictMode>
);
