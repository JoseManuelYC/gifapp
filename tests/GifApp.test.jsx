import { render, screen } from "@testing-library/react";
import { GifApp } from "../src/GifApp";

describe("Pruebas en GifApp", () => {
  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<GifApp />);
    expect(container).toMatchSnapshot();
  });
  test('Debe tener un titulo h1 con el nombre "GifApp"', () => {
    render(<GifApp />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toBe("GifApp");
  });
});
