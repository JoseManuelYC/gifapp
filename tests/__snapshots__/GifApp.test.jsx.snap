// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifApp Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <h1>
      GifApp
    </h1>
    <form
      aria-label="form"
    >
      <input
        placeholder="Buscar un gif"
        type="text"
        value=""
      />
    </form>
    <div
      class="subcontainer"
    >
      <h3>
        Resident Evil
      </h3>
      <h2>
        Cargando...
      </h2>
      <div
        class="card-grid"
      />
    </div>
  </div>
</div>
`;
